src/utils/mtev_lfu.c

```diff
@@ -236,6 +263,12 @@ touch_lfu_cache_no_lock(mtev_lfu_t *c, struct lfu_entry *e)
 mtev_boolean
 mtev_lfu_put(mtev_lfu_t *lfu, const char *key, size_t key_len, void *val)
 {
+  /* a max size of zero means to disable the LFU */
+  if (lfu->max_entries == 0) {
+    /* we don't want calling code to error */
+    return mtev_true;
```

Comment:
Whenever an item gets evicted, the free callback gets invoked. (Aside: might be better to rename this the eviction callback?) Most clients, I think, will have the idea that if an item is successfully inserted into the LFU, it will be possible to retrieve it from the LFU unless it was evicted. A reasonable alternative implementation would be to immediately invoke the free callback if you try to `mtev_lfu_put` and the max size is 0?